http3/server.go

```diff
@@ -436,6 +447,33 @@ func (s *Server) handleConn(conn quic.Connection) error {
 	}).Append(b)
 	str.Write(b)
 
+	var (
+		once   sync.Once
+		mutex  sync.Mutex
+		closed bool
+		// client opened bidirectional stream ID starts at 0 and increase by 4
+		lastID = quic.StreamID(-4)
```

Comment:
```suggestion
		lastID = protocol.InvalidStreamID
```

Reply:
Is it fine to not close quic.Connection on the server side. Right now it's only closed on the server side when an error occurred during request handling.

Reply:
Why don't you want to close it? That's the point of graceful *shutdown* after all, isn't it?

Reply:
Sending goaway frame requires the server having some kinds of references to the quic connection. Closing it is not necessary as stdlib doesn't close either.

Reply:
@marten-seemann  I'm thinking using a counter to track the number of accepted quic connections, and a context that will be canceled when shutdown is called. `context. AfterFunc` can be used to send goaway in this case. The server only knows about the number of the quic connections. What do you think of this method?

Also, perhaps I should implement server side graceful shutdown first, client side later. To keep changes small.